packages/ra-ui-materialui/src/theme/ThemeProvider.tsx

```diff
@@ -46,11 +46,7 @@ export const ThemeProvider = ({
     const themeValue = useMemo(() => {
         try {
             return createTheme(
-                typeof mode === 'object'
-                    ? mode // FIXME: legacy useTheme, to be removed in v5
```

Comment:
I also went ahead and removed this legacy code, as suggested by the comment left here

Reply:
Then you have to add a section in the Upgrade guide to explain what to do for apps using the old useTheme value.